deployment/terraform/resources/functions.tf

```diff
@@ -77,3 +77,29 @@ resource "azurerm_role_assignment" "function-app-animation-container-access" {
     azurerm_function_app.pcfuncs
   ]
 }
+
+resource "azurerm_role_assignment" "function-app-storage-table-data-contributor" {
+  scope                = azurerm_storage_account.pc.id
+  role_definition_name = "Storage Table Data Contributor"
+  principal_id         = azurerm_function_app.pcfuncs.identity[0].principal_id
+
+  depends_on = [
+    azurerm_function_app.pcfuncs
+  ]
+}
+
+data "azurerm_log_analytics_workspace" "log_analytics_workspace" {
+  provider            = azurerm.log_analytics
+  name                = var.log_analytics_workspace_name
+  resource_group_name = var.pc_resources_rg
+}
+
+resource "azurerm_role_assignment" "function-app-log-analytics-access" {
+  scope                = data.azurerm_log_analytics_workspace.log_analytics_workspace.id
+  role_definition_name = "Log Analytics Contributor"
```

Comment:
Is contributor necessary to execute read only queries? If we can have it be a data reader only, that would be preferable.